app/src/main/java/io/intrepid/russell/checkin/CheckInService.java

```diff
@@ -160,33 +167,32 @@ public void onResult(@NonNull Status status) {
     }
 
     private GeofencingRequest getGeofencingRequest() {
+        Geofence.Builder fenceBuilder = new Geofence.Builder()
+                .setTransitionTypes(Geofence.GEOFENCE_TRANSITION_EXIT | Geofence.GEOFENCE_TRANSITION_DWELL)
+                .setExpirationDuration(Geofence.NEVER_EXPIRE)
+                .setLoiteringDelay(LOITERING_DELAY);
+
         return new GeofencingRequest.Builder()
                 .setInitialTrigger(GeofencingRequest.INITIAL_TRIGGER_DWELL | GeofencingRequest.INITIAL_TRIGGER_ENTER)
-                .addGeofence(new Geofence.Builder()
+                .addGeofence(fenceBuilder
                         .setRequestId(LocationData.THIRD_ST.id)
                         .setCircularRegion(LocationData.THIRD_ST.lat, LocationData.THIRD_ST.lng, LocationData.GEOFENCE_RADIUS)
-                        .setTransitionTypes(Geofence.GEOFENCE_TRANSITION_EXIT | Geofence.GEOFENCE_TRANSITION_DWELL)
-                        .setExpirationDuration(Geofence.NEVER_EXPIRE)
-                        .setLoiteringDelay(LOITERING_DELAY)
                         .build())
-                .addGeofence(new Geofence.Builder()
+                .addGeofence(fenceBuilder
                         .setRequestId(LocationData.ROGERS_ST.id)
                         .setCircularRegion(LocationData.ROGERS_ST.lat, LocationData.ROGERS_ST.lng, LocationData.GEOFENCE_RADIUS)
```

Comment:
I'm pretty sure I've seen things which are intended to work in this way. In this case, from looking at the decompiled source, the build() method appears to just read the builder values into the constructor of a Geofence implementation, so it should be fine.

If it's generally a bad pattern I'll try to avoid it in the future. Just trying to clear out some code reuse.